samples/crypto/mbedtls_benchmark/src/benchmark.c

```diff
@@ -0,0 +1,1070 @@
+		if (md_info == NULL) {
+			mbedtls_exit(1);
+		}
+
+		if (mbedtls_hmac_drbg_seed(&hmac_drbg, md_info,
+					   myrand, NULL, NULL, 0) != 0) {
+			mbedtls_exit(1);
+		}
+
+		TIME_AND_TSC("HMAC_DRBG SHA-1 (NOPR)",
+			     mbedtls_hmac_drbg_random(&hmac_drbg, buf,
+						      BUFSIZE));
+
+		if (mbedtls_hmac_drbg_seed(&hmac_drbg, md_info, myrand,
+					   NULL, NULL, 0) != 0) {
+			mbedtls_exit(1);
+		}
+
+		mbedtls_hmac_drbg_set_prediction_resistance(&hmac_drbg,
+						MBEDTLS_HMAC_DRBG_PR_ON);
+
+		TIME_AND_TSC("HMAC_DRBG SHA-1 (PR)",
+			     mbedtls_hmac_drbg_random(&hmac_drbg, buf,
+						      BUFSIZE));
+#endif
+
+#if defined(MBEDTLS_SHA256_C)
+		md_info = mbedtls_md_info_from_type(MBEDTLS_MD_SHA256);
+		if (md_info == NULL) {
+			mbedtls_exit(1);
+		}
+
+		if (mbedtls_hmac_drbg_seed(&hmac_drbg, md_info,
+					   myrand, NULL, NULL, 0) != 0) {
+			mbedtls_exit(1);
+		}
+
+		TIME_AND_TSC("HMAC_DRBG SHA-256 (NOPR)",
+			     mbedtls_hmac_drbg_random(&hmac_drbg, buf,
+						      BUFSIZE));
+
+		if (mbedtls_hmac_drbg_seed(&hmac_drbg, md_info, myrand,
+					   NULL, NULL, 0) != 0) {
+			mbedtls_exit(1);
+		}
+
+		mbedtls_hmac_drbg_set_prediction_resistance(&hmac_drbg,
+					MBEDTLS_HMAC_DRBG_PR_ON);
+
+		TIME_AND_TSC("HMAC_DRBG SHA-256 (PR)",
+			     mbedtls_hmac_drbg_random(&hmac_drbg, buf,
+						      BUFSIZE));
+#endif
+		mbedtls_hmac_drbg_free(&hmac_drbg);
+	}
+#endif
+
+#if defined(MBEDTLS_RSA_C) && defined(MBEDTLS_GENPRIME)
+	if (todo.rsa) {
+		int keysize;
+		mbedtls_rsa_context rsa;
+
+		for (keysize = 2048; keysize <= 4096; keysize *= 2) {
+			snprintk(title, sizeof(title), "RSA-%d",
+				 keysize);
+
+			mbedtls_rsa_init(&rsa, MBEDTLS_RSA_PKCS_V15, 0);
+			mbedtls_rsa_gen_key(&rsa, myrand, NULL, keysize,
+					    65537);
+
+			TIME_PUBLIC(title, " public",
+				    buf[0] = 0;
+				    ret = mbedtls_rsa_public(&rsa, buf, buf));
+
+			TIME_PUBLIC(title, "private",
+				    buf[0] = 0;
+				    ret = mbedtls_rsa_private(&rsa, myrand,
+							      NULL, buf, buf));
+
+			mbedtls_rsa_free(&rsa);
+		}
+	}
+#endif
+
+#if defined(MBEDTLS_DHM_C) && defined(MBEDTLS_BIGNUM_C)
+	if (todo.dhm) {
+		int dhm_sizes[] = {2048, 3072};
+		static const unsigned char dhm_P_2048[] =
+					MBEDTLS_DHM_RFC3526_MODP_2048_P_BIN;
+		static const unsigned char dhm_P_3072[] =
+					MBEDTLS_DHM_RFC3526_MODP_3072_P_BIN;
+		static const unsigned char dhm_G_2048[] =
+					MBEDTLS_DHM_RFC3526_MODP_2048_G_BIN;
+		static const unsigned char dhm_G_3072[] =
+					MBEDTLS_DHM_RFC3526_MODP_3072_G_BIN;
+
+		const unsigned char *dhm_P[] = { dhm_P_2048, dhm_P_3072 };
+		const size_t dhm_P_size[] = { sizeof(dhm_P_2048),
+					      sizeof(dhm_P_3072) };
+		const unsigned char *dhm_G[] = { dhm_G_2048, dhm_G_3072 };
+		const size_t dhm_G_size[] = { sizeof(dhm_G_2048),
+					      sizeof(dhm_G_3072) };
+		mbedtls_dhm_context dhm;
+		size_t olen;
+
+		for (i = 0; (size_t)i <
+				ARRAY_SIZE(dhm_sizes) / sizeof(dhm_sizes[0]);
```

Comment:
A bug slipped in during the rework, no need to divide as `ARRAY_SIZE` is now used.

Reply:
for some reason I was getting these checkpatch errors

-:273: WARNING:ARRAY_SIZE: Prefer ARRAY_SIZE(dhm_sizes)
#273: FILE: samples/crypto/mbedtls_benchmark/src/benchmark.c:822:
+				sizeof(dhm_sizes) / sizeof(dhm_sizes[0]);

- total: 0 errors, 1 warnings, 32 lines checked

Reply:
Sure, the checkpatch warning seems fine. The point is, that when you use `ARRAY_SIZE` macro you should no longer divide by `sizeof(dhm_sizes[0])`, beceause the macro does that for you. So instead:
```
(size_t)i < ARRAY_SIZE(dhm_sizes) / sizeof(dhm_sizes[0]);
```
it should be:
```
(size_t)i < ARRAY_SIZE(dhm_sizes);
```

Reply:
aaah! yes, I will update.